src/http.js

```diff
@@ -32,24 +33,32 @@ const timeout = (promise, ms, abortController) => {
 
   const start = Date.now()
 
+  const timedOut = () => {
+    const time = Date.now() - start
+
+    return time >= ms
+  }
+
   return new Promise((resolve, reject) => {
+    const timeoutID = setTimeout(() => {
+      if (timedOut()) {
+        reject(new TimeoutError())
+        abortController.abort()
+      }
+    }, ms)
+
     const after = (next) => {
       return (res) => {
         clearTimeout(timeoutID)
-        const time = Date.now() - start
 
-        if (time >= ms) {
-          abortController.abort()
+        if (timedOut()) {
```

Comment:
why are we checking for timeout on resolve and reject, shouldn't it be aborted already ?

`after` shoudn't call resolve nor reject if already aborted

Reply:
If the event loop is busy and your timeout is very, very small, it's possible to resolve a promise before the timeout fires, even though for the observer, the amount of time specified for the timeout has passed.  This is mostly evident in weird test behaviour.

Reply:
> `after` shoudn't call resolve nor reject if already aborted

It'll only call reject again if the timeout was reached - we could inspect `abortController.signal.aborted` for completeness but you can't reject a promise twice so it shouldn't hurt.